src/libs/actions/Policy.js

```diff
@@ -355,6 +357,25 @@ function addMembersToWorkspace(invitedEmailsToAccountIDs, welcomeNote, policyID,
     // create onyx data for policy expense chats for each new member
     const membersChats = createPolicyExpenseChats(policyID, invitedEmailsToAccountIDs, betas);
 
+    // Optimistic personal details for the new accounts invited
+    const optimisticPersonalDetails = _.chain(invitedEmailsToAccountIDs)
+        .map(
+            (accountID, memberLogin) =>
+                !_.has(allPersonalDetails, accountID) && [
+                    accountID,
+                    {
+                        accountID,
+                        avatar: UserUtils.getDefaultAvatarURL(accountID),
+                        displayName: LocalePhoneNumber.formatPhoneNumber(memberLogin),
+                        login: OptionsListUtils.addSMSDomainIfPhoneNumber(memberLogin),
+                        isDummy: true,
```

Comment:
Do we actually need `isDummy`? We are able to get the personalDetails to be removed in `clearDeleteMemberError` by `accountID`. Tested locally and seemed to work fine without including this

Reply:
Ah, I guess for the cases where we have an error we want to dismiss but where we don't want to remove that user from the policy details 🤔 

In that case I guess I would call this something a little more descriptive like `isOptimisticDetails` or something like that.

Reply:
Oops, sorry, you're right about that, I mistyped the function name. But the comment still holds, why do we need `isDummy`/`isOptimisticData` if we can remove the optimistically added member by `accountID`?

Reply:
No worries, I'm not explaining things very clearly 😅 Effectively, I'm not sure I understand why we need to include the `isOptimisticData` value at all. 

The logic you have [here](https://github.com/Expensify/App/pull/23157/files#diff-5fd7c80f3a9b43595640753f4037ca2b7d0bebeca557a20c4a06f89dc5028ccfR779-R783) conditionally retrieves the personal details that match the `accountID` and where `isOptimisticData = true`, and then removes those details from `ONYXKEYS.PERSONAL_DETAILS_LIST`, right? 

But whenever we call `clearAddMemberError`, we're going to want to update `ONYXKEYS.PERSONAL_DETAILS_LIST` to remove the details for the provided `accountID`, because that user won't have been added to the workspace successfully. We can do this based on `accountID` alone, so why do we need `isOptimisticData = true` as a condition to be met? 

Hopefully, that's a bit clearer? Let me know if I'm misunderstanding something or if it's still unclear. 

> Can you elaborate on the test you've made?

Yeah, for sure. I just removed [this line](https://github.com/Expensify/App/pull/23157/files#diff-5fd7c80f3a9b43595640753f4037ca2b7d0bebeca557a20c4a06f89dc5028ccfR371) and adjusted [this line](https://github.com/Expensify/App/pull/23157/files#diff-5fd7c80f3a9b43595640753f4037ca2b7d0bebeca557a20c4a06f89dc5028ccfR779) to be as follows: 

```diff
-    if (lodashGet(allPersonalDetails, [accountID, 'isOptimisticData'])) {
+    if (lodashGet(allPersonalDetails, [accountID])) {
```
Then retested using the testing steps for this PR.